### ISSUE TITLE
Notify Admin user when question breaks because of dropped column/altered table

### ISSUE THREAD

**Issue by ignacio-mb** [metabase/metabase#35330]
**Is your feature request related to a problem? Please describe.**
When a change happens at the database level (column dropped, for example) a question that depends on that column breaks. That we know: https://github.com/metabase/metabase/issues/12721

**Describe the solution you'd like**
A simple notification from Metabase saying: This question (or the following questions) don't work because of a change at recognized at the database schema level.
 The errors at the question level is: 
`ERROR: column test_negative_data.value does not exist
  Position: 184`

And we are identifying it in the sync as 
`INFO fields.sync-instances :: Marking Field 'Table 35 'public.test_negative_data' Field 'value'' as inactive.`

**Describe alternatives you've considered**
Parse the logs, db triggers, but it's some work

**How important is this feature to you?**
Requested by a customer, his words: "It's not often that we encounter broken questions, but I do know that if someone drops a column that my query depends on, it's going to break and I wonder when that will happen. Notifying by email when a "Column your question uses has been dropped" or a column's data type has changed, which would result in the next run of the question breaking. I wonder if using some kind of `EXPLAIN` on questions regularly (e.g. every 6-12 hours) so it doesn't actually execute them could help reveal outages."

**Additional context**
https://github.com/metabase/metabase/issues/12721

**Comment by dragonsahead** [comment]
This should be a thing now if using transforms and dependency checks. Right @ignacio-mb ?

**Comment by ignacio-mb** [comment]
That's correct - the specific ask is to have email/slack notifications with info about the broken questions, but I think the Dependency diagnostics page is very sufficient and gives the necessary information for admins to take action:

<img width="1900" height="652" alt="Image" src="https://github.com/user-attachments/assets/ab6830fa-d9f2-438f-bd71-b346691eac02" />

Pro/Enterprise feature
